core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -385,6 +385,23 @@ public static void checkActualAndReferenceFiles() {
         .withConformance(SqlConformanceEnum.LENIENT).ok();
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5507">[CALCITE-5507]
+   * HAVING alias failed when aggregate function in condition</a>. */
+  @Test void testAggregateFunAndAliasInHaving1() {
+    sql("select count(empno) as e\n"
+        + "from emp\n"
+        + "having e > 10 and count(empno) > 10")
+        .withConformance(SqlConformanceEnum.LENIENT).ok();
+  }
+
+  @Test void testAggregateFunAndAliasInHaving2() {
+    sql("select count(empno) as e\n"
+        + "from emp\n"
+        + "having e > 10 or count(empno) < 5")
```

Comment:
I'm not sure if we need to have a different test which has only a minor difference with `testAggregateFunAndAliasInHaving1` (from `AND` to `OR`)

Reply:
I want to show it because one is reduced and another is not reduced.
- `testAggregateFunAndAliasInHaving1` 
`LogicalFilter(condition=[>($0, 10)])` has no `and`.
- `testAggregateFunAndAliasInHaving2`
`LogicalFilter(condition=[SEARCH($0, Sarg[(-∞..5), (10..+∞)])])` has `or`.